lib/db2.js

```diff
@@ -50,6 +51,8 @@ function DB2(settings) {
     ';PORT=' + this.portnumber +
     ';PROTOCOL=' + this.protocol;
 
+  this.connStr = this.dsn || this.connStrGenerate;
+
   this.schema = this.username;
   if (settings.schema) {
     this.connStr += ';CurrentSchema=' + settings.schema;
```

Comment:
@bajtos Should we be doing validation of the DSN to ensure it contains the relevant pieces of information including Schema.  @jannyHou I agree that if using DSN then this block for adding CurrentSchema shouldn't be executed